pcdsdevices/sequencer.py

```diff
@@ -114,6 +116,8 @@ def put_seq(self, sequence, update_length=True):
         self.bd_array.put(seq[1])
         self.fd_array.put(seq[2])
         self.bc_array.put(seq[3])
+        time.sleep(0.5) # dumb sleep to wait for array PVs to reach sequencer
+        self.seq_proc.put(1)
```

Comment:
The only time a `put()` is expected to take a significant amount of time is when "put completion [callbacks]" are used, as it will block until it receives notification from the IOC that it has been written. The components above do not have `put_complete=True`, meaning it defaults to `False`.

Maybe just toss in some debug prints to see which is taking time? (Or enable the `control_layer_log` at the [debug level](https://github.com/bluesky/ophyd/blob/317b41585df9660befc332c829436960606036d7/ophyd/signal.py#L215), but that will be spammy)

Reply:
That's absolutely bizarre

Reply:
Turns out that this totally works. The issue is that camonitor'ing the array PVs is a little misleading. The array PVs were not updating in camonitor until several seconds after the caput, but if you monitored the event codes produced by the sequencer you would see that the sequence was correct. 

This was _not_ the case when the put to SEQ.PROC was not there, so this was indeed a bug in the code. Pushing a cleaned up version of this fix.

Reply:
Makes sense now - so there was just more record logic hiding the fact that the fix actually worked!